include/CIndexStoreDB/CIndexStoreDB.h

```diff
@@ -201,6 +203,30 @@ indexstoredb_index_canonical_symbol_occurences_containing_pattern(
     bool ignoreCase,
     _Nonnull indexstoredb_symbol_occurrence_receiver_t receiver);
 
+/// Gets the set of roles of the passed in symbol relation
+/// @param relation A symbol relation
+INDEXSTOREDB_PUBLIC uint64_t
+indexstoredb_symbol_relation_get_roles(_Nonnull  indexstoredb_symbol_relation_t);
+
+/// Gets the symbol associated with the passed in relation
+/// @param relation A symbol relation
+INDEXSTOREDB_PUBLIC _Nonnull indexstoredb_symbol_t
+indexstoredb_symbol_relation_get_symbol(_Nonnull indexstoredb_symbol_relation_t);
+
+/// Loops through each relation that a passed in symbol has, and performs the passed in function.
+/// The relations are owned by the occurrence and shall not be used after the occurrence is freed.
+/// @param occurrence The symbol occurrence that whose relations should be found.
+/// @param applier The function that should be performed on each symbol relation.
+/// The function should return a boolean indicating whether the looping should continue.
+INDEXSTOREDB_PUBLIC bool
+indexstoredb_symbol_occurrence_relations(_Nonnull indexstoredb_symbol_occurrence_t,
+                                         bool(^ _Nonnull applier)(indexstoredb_symbol_relation_t _Nonnull ));
+
+/// Get the SymbolKind of a Symbol
+/// @param symbol The symbol whose kind should be found.
+INDEXSTOREDB_PUBLIC uint8_t
```

Comment:
I think the return type should change, but I don't know to what. What would the best way be to convert from cpp SymbolKind to Swift SymbolKind?

Here I am returning the raw uint8_t value that underlies the cpp type, but that's not helpful to a swift consumer because they don't know how to convert that to something meaningful (it doesn't look like the raw value corresponds to the equivalent Swift enum).

I could use `getSymbolKindString` and a switch statement to return a String, but is there a better way than that?

Reply:
It's a bit of duplication, but we should really have an `indexstoredb_symbol_kind_t` enum that duplicates the symbol kinds from the lower-level `indexstore_symbol_kind_t`, and then also have a pure Swift implementation using swiftier names.  See for example how we exposed `SymbolRoles` and `indexstoredb_symbol_role_t`.

The return type here can then be `indexstoredb_symbol_kind_t `.